module/PiCamera/pi_cam.py

```diff
@@ -0,0 +1,312 @@
+
+        for k, v in kwargs.items():
+            self.set_param(k, v)
+
+    def __getattr__(self, item):
+        if hasattr(self, item):
+            return self[item]
+        if hasattr(self.camera, item):
+            return self.camera[item]
+
+    def set_iso(self, iso):
+        """The set_iso function is used to store an iso value to the camera.
+        The function call will also show the filtering of ISO values.
+        The V2 Camera has different calculation with grain.
+        Contrary to the V1.3 it follows the ISO film speed standard.
+        Given that it is more likely that different camera's or other ISO readings can be used externally,
+        it is preferable to use a standard rather than a proprietary calculation.
+        source: https://picamera.readthedocs.io/en/latest/api_camera.html?highlight=iso#picamera.PiCamera.iso
+
+        The PiCameraV2
+        :param iso: integer: the new iso value that will be set.
+        The ISO value must be in a possible range, if not the nearest will be selected.
+        :return iso: the actual set value of ISO.
+        the actual return value will always be different due to the grain calculation.
+        In normal circumstances won't be useful. Therefore the return is conditional.
+        """
+
+        possible_iso_range = [100, 200, 320, 400, 500, 640, 800]  # The camera has limited amount of ISO values.
+        if iso not in possible_iso_range:
+            iso = min(possible_iso_range, key=lambda x: abs(x - iso))  # lambda that finds nearest value and sets to iso
+        self.camera.iso = iso * 0.0184  # the multiplication to get the ISO standard grain with v1.3 camera.
+
+    def capture(self):
+        """
+        Function to capture a single frame. Basic implementation of the manual_capture.
+        :return:
+        """
+        self.manual_capture(self.generate_path("pic", ".jpg"))
+
+    def instantiate_resolutions(self):
+        """Instantiating the video resolutions that the PiCam V1.3 supports.
+        Source: https://picamera.readthedocs.io/en/latest/fov.html#sensor-modes"""
+        allowed_video_resolutions = [
+            ['2592', '1944'],
+            ['1296', '972'],
+            ['1296', '730'],
+            ['640', '480'],
+            ['640', '480'],
+            ['1920', '1080'],
+        ]
+        allowed_frame_rates_ranges = [
+            [1, 15],
+            [1, 42],
+            [1, 49],
+            [42.1, 60],
+            [60.1, 90],
+            [1, 30]
+        ]
+
+        for i in range(1, len(allowed_video_resolutions)):
+            param = {
+                'width': allowed_video_resolutions[i][0],
+                'height': allowed_video_resolutions[i][1],
+                'aspect_frame_rate_min': allowed_frame_rates_ranges[i][0],
+                'aspect_frame_rate_max': allowed_frame_rates_ranges[i][1],
+            }
+            self.video_resolutions.append(VideoResolution(param))
+
+    def set_brightness(self, value):
```

Comment:
The members of this class are not marked as private, so you could make them private, or remove the get and set functions.

Reply:
I've assigned the camera as protected given the fact that it's usage is mainly implemented in it's child classes. The setter calls are mainly for interaction with the camera object so I've filtered them for that. I've decided to keep it here, because else our coding style deviates to much from those of other groups.